src/endpointRoutes.js

```diff
@@ -0,0 +1,140 @@
+'use strict'
+
+const KoaBodyParser = require('@viweei/koa-body-parser')
+
+const logger = require('./logger')
+const {handleServerError, validateEndpoint} = require('./util')
+const {
+  deleteEndpoint,
+  createEndpoint,
+  updateEndpoint
+} = require('./db/services/endpoints')
+
+const createEndpointRoute = router => {
+  router.post('/endpoints', async (ctx, next) => {
+    const failureMsg = 'Endpoint creation/update failed: '
```

Comment:
Also with all these messages, the message itself will be easier to identify in the trace if you **start** the message with the verb. ie:

```suggestion
    const failureMsg = 'Create endpoint failed:'
```
Because when you're looking through logs the verbs in the logs help guide you to what happened or what didn't.